### PR TITLE
feat(provider/azure): Add Azure Load Balancer as another type of load balancer than Azure Application Gateway

### PR DESCRIPTION
**Pull request by michaeljqzq**
* Add a selector for load balancer type (Azure Load Balancer / Azure Application Gateway) when creating load balancer
* Show load balancer type in load balancer details
* Other changes in server group configuration related to load balancer

### CODE REVIEW

**Comment by michaeljqzq** [comment]
This PR is linked with https://github.com/spinnaker/clouddriver/pull/3646

**Comment by duftler** [comment]
Hoping to get this in for 1.14.

File: app/scripts/modules/azure/loadBalancer/details/loadBalancerDetail.controller.js [hunk 1]
```diff
@@ -91,6 +91,17 @@ module.exports = angular
                 });
                 $scope.securityGroups = _.sortBy(securityGroups, 'name');
               }
+
+              if ($scope.loadBalancer.loadBalancerType) {
+                let type = $scope.loadBalancer.loadBalancerType;
```
Comment (maggieneterval) [inline app/scripts/modules/azure/loadBalancer/details/loadBalancerDetail.controller.js]: can this be a `const`?

File: app/scripts/modules/azure/loadBalancer/configure/AzureLoadBalancerChoiceModal.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  IAzureLoadBalancerChoiceModalState
+> {
+  private static parentProps: ILoadBalancerModalProps;
+  public static defaultProps: Partial<ILoadBalancerModalProps> = {
+    closeModal: noop,
+    dismissModal: noop,
+  };
+
+  public static show(props: ILoadBalancerModalProps): Promise<void> {
+    this.parentProps = props;
```
Comment (maggieneterval) [inline app/scripts/modules/azure/loadBalancer/configure/AzureLoadBalancerChoiceModal.tsx]: What is the purpose of `parentProps`?

**Comment by maggieneterval** [comment]
Hey @michaeljqzq left a few comments, should be able to get this in before we cut the 1.14 branch.

File: app/scripts/modules/azure/loadBalancer/configure/AzureLoadBalancerChoiceModal.tsx (hunk 2 above)
Comment (michaeljqzq) [reply app/scripts/modules/azure/loadBalancer/configure/AzureLoadBalancerChoiceModal.tsx]: Removed it and use this.props.app to inject application property

**Pull request merged**